### PR TITLE
DAOS-5866 test: Negative testing (OSA dmg command)

### PR DESCRIPTION
**Pull request by byron-marohn**
DAOS-5866 test: Negative testing (OSA dmg command) 
    Test-tag-hw-medium: pr,hw,medium,ib2 osa
    
    Summary:
    - Pass invalid values to dmg pool exclude[reintegrate, drain] commands.
    - Check whether dmg provides appropriate errors.
    - Extend code is added but commented (due to DAOS-6838 issue)

This change was already landed to master, cherry pick for 1.2

### CODE REVIEW

**Review by shimizukko — DISMISSED** [review]
LGTM

**Review by rpadma2 — APPROVED** [review]

**Comment by rpadma2** [comment]
Added the test-tag to run the OSA tests. Test-tag-hw-medium: pr,hw,medium,ib2 osa.

**Review by rpadma2 — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]
LGTM
hmm I don't see the diff from the last time I reviewed. I see the entire change instead.
I know you're skipping it, but did you run without extend case so that the code works?

File: src/tests/ftest/osa/osa_dmg_negative_test.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            self.start_additional_servers(self.extra_servers)
+
+        # Get rank, target from the test_dmg_sequence
+        # Some test_dmg_sequence data will be invalid, valid.
+        for val in range(0, num_pool):
+            for i in range(len(self.test_seq)):
+                self.pool = pool[val]
+                if extend is True:
+                    scm_size = self.pool.scm_size
+                    nvme_size = self.pool.nvme_size
```
[the remark this answers is not in the document]
Comment (rpadma2) [reply src/tests/ftest/osa/osa_dmg_negative_test.py:90]: Yes, I ran it manually. With extend functionality not working, this problem might be hidden... Let's just do cherry-pick for now and fix the issue when we enable extend testing.

**Comment by rpadma2** [comment]
> LGTM
> hmm I don't see the diff from the last time I reviewed. I see the entire change instead.
> I know you're skipping it, but did you run without extend case so that the code works?

There is no difference... All, I did is to run the existing OSA tests with this PR (just did git commit --amend to add test-tag to run OSA tests).

**Comment by rpadma2** [comment]
io.LlnlMpi4py, romio issues are known problem.

File: src/tests/ftest/osa/osa_dmg_negative_test.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    Test Class Description: This test
+    passes invalid parameters to dmg OSA
+    related commands and check whether
+    appropriate errors are returned.
+
+    :avocado: recursive
+    """
+    def setUp(self):
+        """Set up for test case."""
+        super(OSADmgNegativeTest, self).setUp()
```
[the remark this answers is not in the document]
Comment (rpadma2) [reply src/tests/ftest/osa/osa_dmg_negative_test.py:24]: python3 issues will not be fixed now.

**Comment by rpadma2** [comment]
poolcreatetest failure is related to https://jira.hpdd.intel.com/browse/DAOS-6784

**Comment by rpadma2** [comment]
erasure code failure (daos_agent start failure) is related to DAOS-6790

**Comment by rpadma2** [comment]
This PR can be merged. All failures have tickets (DAOS-6905,DAOS-6790, DAOS-6784).

**Pull request merged**